### PR TITLE
Threads: Thread Interrupting

### PR DESCRIPTION
**Pull request by denisdimitrov**

### CODE REVIEW

**Comment by EmilGeorgiev** [comment]
Is this second task?

File: src/main/java/com/clouway/InterruptThread.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package com.clouway;
+
+import static java.lang.Thread.currentThread;
+
+/**
+ * Created by clouway on 16.09.16.
+ *
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public class InterruptThread {
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/InterruptThread.java]: For classes that acts as a an entry point of the application (class that have ```main``` method),  we can transform its name by ```name of application + Bootsrap```. For example here we can name this class ```CounterBootstrap``` or ```TwoCounterBootstrap```

File: src/main/java/com/clouway/InterruptThread.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    private static class Threads implements Runnable {
+        public void run() {
+            Thread firstThread = new Thread(new Threads());
+            Thread secondThread = new Thread(new Threads());
+            int counter = 0;
+            int firstThreadLifetime = 5;
+            int secondThreadLifetime = 12;
+            while (true) {
+                if (counter <= firstThreadLifetime | counter <= secondThreadLifetime) {
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/InterruptThread.java]: you can use common flag for each thread as ```AtomicBoolean```. And when first threads finish set this flag to ```false``` and another threads will see that and stop its execution.

File: src/main/java/com/clouway/InterruptThread.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * Created by clouway on 16.09.16.
+ *
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public class InterruptThread {
+
+    private static class Threads implements Runnable {
+        public void run() {
+            Thread firstThread = new Thread(new Threads());
+            Thread secondThread = new Thread(new Threads());
```
Comment (denisdimitrov) [inline src/main/java/com/clouway/InterruptThread.java]: Fixed.

File: src/main/java/com/clouway/InterruptThread.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import static java.lang.Thread.currentThread;
+
+/**
+ * Created by clouway on 16.09.16.
+ *
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public class InterruptThread {
+
+    private static class Threads implements Runnable {
```
Comment (denisdimitrov) [inline src/main/java/com/clouway/InterruptThread.java]: Fixed.

File: src/main/java/com/clouway/InterruptThread.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                        counter++;
+                        firstThread.sleep(500);
+                        secondThread.sleep(500);
+                        System.out.println(currentThread().getName() + " " + counter);
+                    } catch (InterruptedException z) {
+                        System.out.println(currentThread().getName() + " got interrupted!");
+                    }
+                    if (counter >= firstThreadLifetime | counter >= secondThreadLifetime) {
+                        break;
+                    }
```
Comment (denisdimitrov) [inline src/main/java/com/clouway/InterruptThread.java]: The if statement at the bottom with break inside.

File: src/main/java/com/clouway/InterruptThread.java (hunk 1 above)
Comment (denisdimitrov) [inline src/main/java/com/clouway/InterruptThread.java]: Fixed.

File: src/main/java/com/clouway/Counter.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+/**
+ * Created by clouway on 26.09.16.
+ *
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public class Counter implements Runnable {
+    private int seconds;
+    private int counter;
+    private Counter counterBootstrap;
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/Counter.java]: Actually this is not used, you only check whether ```Counter``` is null. If this is not used you can removed it.

Comment (denisdimitrov) [inline src/main/java/com/clouway/Counter.java]: I'm using it in the setCounter method.

Comment (EmilGeorgiev) [inline src/main/java/com/clouway/Counter.java]: In set method you initialize ```Counter```, but it is not used. If we follow this logic we can create next setter mettods.
```java
puvlic class Counter implements Runnable {
    private Counter counterBootstrap;
    private String address;
    private String name;
    private String lastName;
    private Double salary ;
    private Integer age;
    private Double price;

    public void setAddress(..){...}
    public void setName(..){...}
    public void setLastName(..){...}
    public void setSalary(..){...}
    public void setSalary(..){...}
    public void setPrice(..){...}
}
```

All set methods are just as useful for your class as ```setCounter()```.

Comment (denisdimitrov) [inline src/main/java/com/clouway/Counter.java]: Yup, I saw my mistake. I just removed it.

**Comment by EmilGeorgiev** [comment]
LGTM

**Pull request merged**